src/sagemaker/serve/builder/model_builder.py

```diff
@@ -235,6 +237,9 @@ class ModelBuilder(Triton, DJL, JumpStart, TGI, Transformers):
     model_server: Optional[ModelServer] = field(
         default=None, metadata={"help": "Define the model server to deploy to."}
     )
+    model_metadata: Optional[Dict[str, str]] = field(
```

Comment:
Minor wdyt think  about this being: Dict[str, Any] to account for False and Numbers?

https://docs.python.org/3/library/typing.html#the-any-type

We want to cover 
etc.
- HF_OPTIMUM_BATCH_SIZE
- HF_TRUST_REMOTE_CODE

https://github.com/aws/sagemaker-huggingface-inference-toolkit

Reply:
Can we also add documentation here, that HF Task can be overridden here

Reply:
The toolkit is treating these values as strings as well
`HF_TRUST_REMOTE_CODE="True"`
`HF_OPTIMUM_BATCH_SIZE="1"`
Either way I think we should explicitly tell the customers what to expect using this field